doc/pages/architecture/080-mqtt-v1-protocol.md

```diff
@@ -0,0 +1,278 @@
+
+## Publishing Data
+
+Either Astarte or a device might publish new data on a interface/endpoint specific topic.
+The topic is built using `/<interface name>/<path>` schema, and it is used regardless the type of interface or mapping.
+
+Also `/` path is a valid path for object aggregated interfaces.
+
+Following device topics are valid:
+
+* `/com.example.MyInterface/some/path`
+* `/org.example.DraftInterface/otherPath`
+* `/com.example.astarte.ObjectAggregatedInterface/`
+
+Data messages QoS is chosen according to mapping settings, such as reliability. Properties are always published using QoS 2.
+
+| Interface Type | Reliability     | QoS |
+|----------------|-----------------|-----|
+| properties     | always unique   | 2   |
+| datastream     | unreliable      | 0   |
+| datastream     | guaranteed      | 1   |
+| datastream     | unique          | 2   |
+
+### Payload Format
+
+Payload format might change according to the message type. Payloads are always BSON encoded, except for unset messages that are empty.
+
+#### Property Message
+
+Property messages have a "v" key (which means value) and they look like the followings:
+
+* `{"v": "string property value"}`
+* `{"v": 10}`
+* `{"v": true}`
+
+Previous payloads are in order BSON encoded as following hex dumps:
+
+```
+0000000 22 00 00 00 02 76 00 16 00 00 00 73 74 72 69 6e
+0000020 67 20 70 72 6f 70 65 72 74 79 20 76 61 6c 75 65
+0000040 00 00
+```
+
+```
+0000000 0c 00 00 00 10 76 00 0a 00 00 00 00
+```
+
+```
+0000000 09 00 00 00 08 76 00 01 00
+```
+
+Property messages order must be preserved and they must be consumed in order. The same property with the same value can be sent several times, this behavior is allowed but discouraged: it's up to the device to avoid useless messages.
+A device must also make sure to publish all the properties that have been changed while the device was offline.
+
+#### Unset Property Message
+
+Properties can be unset with an unset message. An unset message is just an empty 0 bytes payload.
+
+#### Datastream Message (individual aggregation)
+
+Datastream messages for interfaces with individual aggregation have a "v" key and an optional "t" key (which means timestamp) and they look like the followings:
+
+* `{"v": false}`
+
+* `{"v": 16.73}`
+
+* `{"v": 16.73, "t": 1537449422890}`
+
+Timestamps are UTC timestamps (BSON 0x09 type), when not provided reception timestamp is used.
+
+Previous payloads are in order BSON encoded as following hex dumps:
+
+```
+0000000 09 00 00 00 08 76 00 00 00
+```
+
+```
+0000000 10 00 00 00 01 76 00 7b 14 ae 47 e1 ba 30 40 00
+```
+
+```
+0000000 1b 00 00 00 09 74 00 2a 70 20 f7 65 01 00 00 01
+0000020 76 00 7b 14 ae 47 e1 ba 30 40 00
+```
+
+#### Datastream Message (object aggregation)
+
+Datastream messages (for interfaces with object aggregation) have a "v" key, that is a BSON subdocument, and an optional "t" key (which means timestamp) and they look like the following:
```

Comment:
Rephrase, you specified this before:
Datastream messages for interfaces with object aggregation [support every Astarte payload standard field](#astarte-payload-standard-field), but in this case *value* is a BSON subdocument, in which each key represent an individual mapping of the aggregation. Valid examples are: